controllers/util/common/common.go

```diff
@@ -0,0 +1,208 @@
+	found := &rbacv1.ClusterRoleBinding{}
+	err := ch.Get(context.TODO(), types.NamespacedName{
+		Name:      crb.Name,
+		Namespace: crb.Namespace,
+	}, found)
+	return found, err
+}
+
+func (ch *ControllerHelper) DeleteClusterRoles(clusterRoles []*rbacv1.ClusterRole) error {
+	logger := ch.Log.WithName("DeleteClusterRoles")
+	for _, cr := range clusterRoles {
+		found, err := ch.getClusterRole(cr)
+		if err != nil && errors.IsNotFound(err) {
+			continue
+		} else if err != nil {
+			logger.Error(err, "failed to get ClusterRole", "Name", cr.GetName())
+			return err
+		} else {
+			logger.Info("deleting ClusterRole", "Name", cr.GetName())
+			if err := ch.Delete(context.TODO(), found); err != nil {
+				logger.Error(err, "failed to delete ClusterRole", "Name", cr.GetName())
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (ch *ControllerHelper) ReconcileClusterRole(clusterRoles []*rbacv1.ClusterRole) error {
+	logger := ch.Log.WithValues("Resource Type", "ClusterRole")
+	for _, cr := range clusterRoles {
+		_, err := ch.getClusterRole(cr)
+		if err != nil && errors.IsNotFound(err) {
+			logger.Info("Creating a new ClusterRole", "Name", cr.GetName())
+			err = ch.Create(context.TODO(), cr)
+			if err != nil {
+				return err
+			}
+		} else if err != nil {
+			logger.Error(err, "Failed to get ClusterRole", "Name", cr.GetName())
+			return err
+		} else {
+			err = ch.Update(context.TODO(), cr)
+			if err != nil {
+				logger.Error(err, "Failed to update ClusterRole", "Name", cr.GetName())
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (ch *ControllerHelper) getClusterRole(cr *rbacv1.ClusterRole) (*rbacv1.ClusterRole, error) {
+	found := &rbacv1.ClusterRole{}
+	err := ch.Get(context.TODO(), types.NamespacedName{
+		Name:      cr.GetName(),
+		Namespace: cr.GetNamespace(),
+	}, found)
+	return found, err
+}
+
+func (ch *ControllerHelper) HasFinalizer(instance crutils.Instance) (bool, error) {
+	accessor, finalizerName, err := ch.getAccessorAndFinalizerName(instance)
+	if err != nil {
+		return false, err
+	}
+
+	return util.Contains(accessor.GetFinalizers(), finalizerName), nil
+}
+
+func (ch *ControllerHelper) AddFinalizerIfNotPresent(instance crutils.Instance,
+	unwrappedInstance client.Object) error {
+	logger := ch.Log.WithName("AddFinalizerIfNotPresent")
+
+	accessor, finalizerName, err := ch.getAccessorAndFinalizerName(instance)
+	if err != nil {
+		return err
+	}
+
+	if !util.Contains(accessor.GetFinalizers(), finalizerName) {
+		logger.Info("adding", "finalizer", finalizerName, "on", accessor.GetName())
+		accessor.SetFinalizers(append(accessor.GetFinalizers(), finalizerName))
+
+		if err := ch.Update(context.TODO(), unwrappedInstance); err != nil {
+			logger.Error(err, "failed to add", "finalizer", finalizerName, "on", accessor.GetName())
+			return err
+		}
+	}
+	return nil
+}
+
+func (ch *ControllerHelper) RemoveFinalizer(instance crutils.Instance,
+	unwrappedInstance client.Object) error {
```

Comment:
same as [this](https://github.com/IBM/ibm-block-csi-operator/pull/258/files?diff=split&w=1#r895710928)

Reply:
same answer